infra/terraform/global/iam.tf

```diff
@@ -0,0 +1,27 @@
+resource "aws_iam_user" "auth0_ses_user" {
+    name = "auth0-ses-user"
+}
+
+resource "aws_iam_user_policy" "auth0-ses-user" {
```

Comment:
I see your point about the Terraform resource itself as you refer to it using the resource type as well.
Still, may be the case of use a more specific name for the identifier in AWS (`name = "auth0-ses-user"` bit).

As said, functionally looks good so feel free to ignore these comments on naming if you want.